### ISSUE TITLE
Does it improve performance if we change the default "re" engine to "oniguruma"

### ISSUE THREAD

**Issue by felixhao28** [pygments/pygments#1798]
Oniguruma has always been used in TextMate/VSCode lexers due to the advantage of better performance. Does it also apply to pygments? Can we add an option to provide a custom regex engine to override the default "re" engine?

**Comment by Anteru** [comment]
Possibly -- this would require some proper way to benchmark (probably by profiling all test cases and see how long that takes to match), and then see if it's worth replacing. Does Onigurama support Python and has a way to timeout reliably? That would be a major feature that the Python built-in regex implementation is missing.

**Comment by birkenfeld** [comment]
The most logical candidate is the "regex" library though. Not sure about timeout there either.

**Comment by Anteru** [comment]
Agreed, but if someone is willing to put in new infrastructure to allow changing the regex library, might as well support more than one. This could be an "extra" version.

**Comment by Moosems** [comment]
I could swear there was another issue like this for another regex library but after searching every keyword that came to mind and combing through a lot of issues I can't find it. Is it deja vu or does anyone else know what I'm referencing?

**Comment by Moosems** [comment]
Not the one I remember but related: https://github.com/pygments/pygments/issues/1425#issuecomment-613231591
